### DISCUSSION TITLE
Authenticating with multiple different providers simultaneously

### DISCUSSION THREAD

**Discussion by Kevin-McGonigle** [nextauthjs/next-auth#7353] [Help]
I've been going through the NextAuth.js issues and discussions recently and have seen a lot of interest in simultaneously maintaining multiple sessions with distinct providers for a single user.

The use cases for this are pretty extensive. For example, suppose I would like to support sign-in to my application using Auth0, but I would also like to authenticate with Dropbox so that I can interact with the user's files. Or, suppose I would like to similarly authenticate with Dropbox and Google so that I can move a user's files from one service to the other. It's pretty clear that this functionality would unlock a wide variety of possibilities.

My case is similar to my first example, and I've been trying to find a graceful way to accomplish this with NextAuth.js with very little luck. There seem to be some uber-clever individuals that have figured it out, but mostly through some pretty gnarly workarounds and almost exclusively involving a database - both of which I'm hoping to avoid.

So, the big questions: Is there a way to use NextAuth.js in its current state to authenticate with multiple providers simultaneously? And, if not, is support for this something that could be added - considering the demand?

**Comment by BitBuf** [comment]
Bumping this - have you managed to find a workaround as of now?
